src/test/java/ch/uzh/ifi/hase/soprafs21/service/ChatServiceTest.java

```diff
@@ -0,0 +1,108 @@
+package ch.uzh.ifi.hase.soprafs21.service;
+
+import ch.uzh.ifi.hase.soprafs21.entity.ChatMessage;
+import ch.uzh.ifi.hase.soprafs21.entity.Conversation;
+import ch.uzh.ifi.hase.soprafs21.entity.User;
+import ch.uzh.ifi.hase.soprafs21.repository.ChatMessageRepository;
+import ch.uzh.ifi.hase.soprafs21.repository.ConversationRepository;
+import ch.uzh.ifi.hase.soprafs21.repository.UserRepository;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.jdbc.Sql;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@SpringBootTest
+@Sql(value = {"/data_init.sql"})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class ChatServiceTest {
+
+    @Autowired
+    private ChatService chatService;
+
+    @Autowired
+    private ChatMessageRepository chatMessageRepository;
+
+    @Autowired
+    private ConversationRepository conversationRepository;
+
+    @Mock
+    private Authentication authenticationMock;
+
+    @Mock
+    private SecurityContext securityContextMock;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @BeforeEach
+    void setUp() {
+        Mockito.when(authenticationMock.getPrincipal()).thenReturn("mark@twen.de");
+        Mockito.when(securityContextMock.getAuthentication()).thenReturn(authenticationMock);
+        SecurityContextHolder.setContext(securityContextMock);
+    }
+
+    @Test
+    void getAllConversations() {
+        List<Conversation> conversations = chatService.getAllConversations();
+        assertEquals(2, conversations.size());
+        assertEquals(5, conversations.get(0).getMessages().size());
+        LocalDateTime currentDateTime = LocalDateTime.now();
+
+        for(ChatMessage chatMessage: conversations.get(0).getMessages()) {
+            assertTrue(currentDateTime.isAfter(chatMessage.getTimeStamp()));
+            currentDateTime = chatMessage.getTimeStamp();
+        }
+    }
```

Comment:
Used only for the integration test. What do you mean by "set up that test data for tests only"?

Reply:
I was not sure if the data in the sql file a) gets loaded into the database after starting the application (regardless of whether it's running tests, started as a dev server, or started in production) or b) is only used when running tests. If it's a), I was saying that it might be better to not rely on data defined outside the tests in our test cases, i.e. that the tests should create all the data they base their assertions on themselves in a setup method or similar, not that our application *always* populates the DB (regardless of whether it's run in production, for the tests, ...) with some data that is then used for testing.

Reply:
I see. It is not "data.sql" it is "data_init.sql", so it won't be loaded automatically on starting the application. I can rename it to something else not to create confusion.

Reply:
The confusion was all mine, you can leave it like this